Libraries/LibWeb/Crypto/SubtleCrypto.cpp

```diff
@@ -707,25 +707,28 @@ JS::ThrowCompletionOr<GC::Ref<WebIDL::Promise>> SubtleCrypto::derive_key(Algorit
         }
 
         // 15. Let result be the result of performing the import key operation specified by normalizedDerivedKeyAlgorithmImport using "raw" as format, secret as keyData, derivedKeyType as algorithm and using extractable and usages.
-        auto result = normalized_derived_key_algorithm_import.methods->import_key(*normalized_derived_key_algorithm_import.parameter, Bindings::KeyFormat::Raw, secret.release_value()->buffer(), extractable, key_usages);
-        if (result.is_error()) {
-            WebIDL::reject_promise(realm, promise, Bindings::dom_exception_to_throw_completion(realm.vm(), result.release_error()).release_value().value());
+        auto result_or_error = normalized_derived_key_algorithm_import.methods->import_key(*normalized_derived_key_algorithm_import.parameter, Bindings::KeyFormat::Raw, secret.release_value()->buffer(), extractable, key_usages);
+        if (result_or_error.is_error()) {
+            WebIDL::reject_promise(realm, promise, Bindings::dom_exception_to_throw_completion(realm.vm(), result_or_error.release_error()).release_value().value());
             return;
         }
+        auto result = result_or_error.release_value();
 
         // 16. If the [[type]] internal slot of result is "secret" or "private" and usages is empty, then throw a SyntaxError.
-        if ((result.release_value()->type() == Bindings::KeyType::Secret || result.release_value()->type() == Bindings::KeyType::Private) && key_usages.is_empty()) {
+        if ((result->type() == Bindings::KeyType::Secret || result->type() == Bindings::KeyType::Private) && key_usages.is_empty()) {
             WebIDL::reject_promise(realm, promise, WebIDL::SyntaxError::create(realm, "usages must not be empty"_string));
             return;
         }
 
-        // AD-HOC: Set the [[extractable]] internal slot of key to be extractable.
-        //         See: https://github.com/w3c/webcrypto/issues/383
-        auto key = result.release_value();
-        key->set_extractable(extractable);
+        // 17. Set the [[extractable]] internal slot of result to extractable.
+        result->set_extractable(extractable);
 
-        // 17. Resolve promise with result.
-        WebIDL::resolve_promise(realm, promise, key);
+        // 18. Set the [[usages]] internal slot of result to the normalized value of usages.
+        normalize_key_usages(key_usages);
```

Comment:
it is somewhat disturbing to see here a spooky action at a distance on parameters here; won’t this be an issue?

Reply:
Yeah, it's probably better to sort the target array instead since that's a copy anyway. This code was copied verbatim so there are two places where you could update this. You want to PR it?

Reply:
i appreciate your inclusiveness, yet i’ll pass, at least on this one. 🎩